libsplinter/src/biome/rest_api/actix/user.rs

```diff
@@ -277,69 +277,35 @@ fn add_delete_user_method(
                 .into_future(),
         );
     };
-    Box::new(into_bytes(payload).and_then(move |bytes| {
-        let username_password = match serde_json::from_slice::<UsernamePassword>(&bytes) {
-            Ok(val) => val,
+    match authorize_user(&request, &user_id, &secret_manager, &rest_config) {
+        AuthorizationResult::Authorized => match user_store.remove_user(&user_id) {
+            Ok(()) => Box::new(
+                HttpResponse::Ok()
+                    .json(json!({ "message": "User deleted sucessfully" }))
+                    .into_future(),
+            ),
             Err(err) => {
```

Comment:
You should probably check what error is returned. Currently if the user does not exists in the database (for example it has already been deleted) the REST API is returning a 500 error. It should probably return a 404.